### PR TITLE
Improve logic to display "One time shipping" label on Product Details

### LINKED ISSUE #12785: Inconsistent Shipping details in product detail screen for subscription products
**Describe the bug**
When enabling `One time Shipping` for a subscription product, if we then set a `Subscription free trial` the one time shipping option should automatically be disabled. However, it will be shown as enabled in the product detail screen and disabled in the Shipping details screen. Se the scree recording:

https://github.com/user-attachments/assets/83150b88-83cd-450f-89ac-6fc52f2ea995

**To Reproduce**

Follow the screen recording instructions above to reproduce the bug. 

**Expected behavior**
The product detail screen should reflect the new disabled state for `One-time Shipping` as soon as the subscription free trial is enabled.

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #12785

### Description

This PR deals with the ["One time shipping" feature in Subscription Product](https://woocommerce.com/document/subscriptions/creating-subscription-products/#one-time-shipping) and the way its state is displayed in Product Details screen.

The documentation states this:
> Note: For shipping to be charged on initial order, the subscription must not have a free trial or be synchronized to a date in the future. The One-Time Shipping option will be disabled if your product has a free trial or is synchronized.

This PR updates the Android behavior to match iOS:
1. If free trial/future sync is enabled, then we should not display the "One time shipping" description label in Product details, as it is disabled automatically
2. If there is no free trial/future sync and the feature is enabled inside "Shipping", then we want to display the "One time shipping: enabled" description label. If the feature is disabled, we should not display the "One time shipping" description label.
3. Changes in the free trial and toggles should be reflected in the Product Details area correclty.

### Steps to reproduce

1. Start creation for Simple Subscription product in the app, 
2. Fill in the name, price, then tap **Add More Details** and select **Shipping**
4. Fill in just the **Weight** value, then press back,
5. Ensure the **Shipping** card in Product Details does not mention "One time shipping",
6. Go back to **Shipping** and enable **One time shipping**, then press back,
7. Ensure the **Shipping** card in Product Details says "One time shipping:  Enabled",
8. Tap **Subscription free trial** and enter a random number (e.g: 3 days), then press back,
9. Check **Shipping** card and ensure it does not say anything about "One time shipping",
10. Tap **Shipping** and ensure **One Time Shipping** toggle is now disabled (this is an existing behavior and not part of this PR, but good to check) 

Optionally, also test this with an existing Product. The behavior should be the same.

### Testing information

I tested this in an emulator with both new simple subscription product and existing product.

### The tests that have been performed

Everything in the list above.

### Images/gif

Expected states:
| When there is free trial | When no free trial and toggle enabled | When no free trial and toggle disabled |
|-|-|-|
| ❌  label | ✅  label | ❌  label 
| Screenshot_20241128_154311 | Screenshot_20241128_154340 |  Screenshot_20241128_154356 | 

https://github.com/user-attachments/assets/0e8959ce-120b-4740-af54-c649b4e44273

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilder.kt [hunk 1]
```diff
@@ -533,6 +529,23 @@ class ProductDetailCardBuilder(
...
+    // In all other cases (not supported, or supported but disabled), we show nothing.
+    private fun buildOneTimeShippingDescription(subscription: SubscriptionDetails?): String {
+        if (subscription == null) return ""
+
+        return when {
+            subscription.supportsOneTimeShipping && subscription.oneTimeShipping ->
+                resources.getString(string.subscription_one_time_shipping_enabled)
+            else -> ""
+        }
+    }
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilder.kt]: We can simplify this logic as follows (I just merged two conditions that ends with `""`):
```suggestion
private fun buildOneTimeShippingDescription(subscription: SubscriptionDetails?) =
        if (subscription != null && subscription.supportsOneTimeShipping && subscription.oneTimeShipping) {
            resources.getString(string.subscription_one_time_shipping_enabled)
        } else {
            ""
        }
```

**Review by irfano — APPROVED** [review]
Thank you for clearly explaining the rationale behind the change. LGTM and works as expected. 👍🏻 I’m approving but not merging to give you a chance to consider my suggestion regarding code style.
I also noticed another issue with the "One time shipping" label: https://github.com/woocommerce/woocommerce-android/issues/13027. It's reproducable on `trunk` and outside the scope of this PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilder.kt (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/details/ProductDetailCardBuilder.kt]: Thanks! Good point there.

**Comment by hafizrahman** [comment]
> I also noticed another issue with the "One time shipping" label: https://github.com/woocommerce/woocommerce-android/issues/13027. It's reproducable on trunk and outside the scope of this PR.

Thanks! I'll take a look at that issue next.

**Pull request merged**